apps/api/src/donations/donations.service.ts

```diff
@@ -120,38 +120,65 @@ export class DonationsService {
     const campaign = await this.campaignService.validateCampaignId(sessionDto.campaignId)
     const { mode } = sessionDto
     const appUrl = this.config.get<string>('APP_URL')
-    const metadata: DonationMetadata = { campaignId: sessionDto.campaignId }
+    const metadata: DonationMetadata = {
+      campaignId: sessionDto.campaignId,
+      personId: sessionDto.personId,
+    }
+
+    const items = await this.prepareSessionItems(sessionDto, campaign)
 
-    const session = await this.stripeClient.checkout.sessions.create({
+    const createSessionRequest: Stripe.Checkout.SessionCreateParams = {
       mode,
       customer_email: sessionDto.personEmail,
-      line_items: this.prepareSessionItems(sessionDto, campaign),
+      line_items: items,
       payment_method_types: ['card'],
-      payment_intent_data: mode === 'payment' ? { metadata } : undefined,
-      subscription_data: mode === 'subscription' ? { metadata } : undefined,
+      payment_intent_data: mode == 'payment' ? { metadata } : undefined,
+      subscription_data: mode == 'subscription' ? { metadata } : undefined,
       success_url: sessionDto.successUrl ?? `${appUrl}/success`,
       cancel_url: sessionDto.cancelUrl ?? `${appUrl}/canceled`,
       tax_id_collection: { enabled: true },
+    }
+
+    Logger.debug('[ CreateCheckoutSession ]', createSessionRequest)
+
+    const session = await this.stripeClient.checkout.sessions.create(createSessionRequest)
+
+    Logger.log('[ CreateInitialDonation ]', {
+      session: session,
     })
 
-    await this.createInitialDonation(campaign, sessionDto, session.payment_intent as string)
+    await this.createInitialDonation(
+      campaign,
+      sessionDto,
+      (session.payment_intent as string) ?? session.id,
```

Comment:
I'm not sure the default `session.id` should be added here instead of `paymentIntentId` when `session.payment_intent` is not present.

This might lead to storing different data in the same db column.

Reply:
well.. I am open to suggestions...
Initially I have skipped creating "initial" donations for subscriptions.
@igoychev convinced me that it is better to have some track record that a payment has been attempted. 
When the actually payment is received it will be attached and the fields will be properly populated.

Reply:
@kachar we already use the extPaymentIntentId to hold different data - for example for bank payments it holds the bank DocumentReferenceId. The safety comes from the extPaymentMethodId which indicates what type of reference this is - currently card or bank. 

So to make it safe for subscriptions too, one proposal is to set extPaymentMethodId to 'stripe-subscription'.

Reply:
We are already doing that:

        extPaymentMethodId: sessionDto.mode === 'subscription' ?  'subscription' : 'card',

Reply:
Sounds good. I can try to prepare a new PR for this.
Will check during the weekend.

Reply:
Another thing that was brought to my attention is that `extPaymentMethodId` column expects the stripe expects an id like `pm_1MEzEO2eZvKYlo2CPlygE9a3` of the actual card that has done the payment. As I see we place hardcoded strings like `subscription` or `card` which is not correct in terms of data architecture.